postprocessor/src/main/java/com/azure/autorest/postprocessor/implementation/CodeFormatterUtil.java

```diff
@@ -44,6 +47,7 @@ public final class CodeFormatterUtil {
      * @throws Exception If code formatting fails.
      */
     public static void formatCode(Map<String, String> files, NewPlugin plugin) throws Exception {
+        AtomicReference<Logger> loggerReference = new AtomicReference<>();
```

Comment:
Why do we use `AtomicReference` here?

Reply:
We have `files.entrySet().parallelStream().forEach` below. Here is just a trick to try set the Logger var only once (and no set at all if none used).